### ISSUE TITLE
YatNMN: `constant_alpha=False` silently coerced to constant α=0.0 (zeros the layer output)

### ISSUE THREAD

**Issue by mlnomadpy** [azettaai/nmn#3]
## Summary

`nmn.nnx.layers.YatNMN.__init__` accepts `constant_alpha: Optional[Union[bool, float]] = None`, but the branch handling falsy non-`None` values silently coerces `False → float(False) → 0.0`, making α a constant 0 and zeroing the layer output. Gradients also can't flow back through the YatNMN's kernel/bias because `d(y·0)/dθ = 0`, so the layer becomes a permanent no-op without any warning.

Easy to hit if a user passes a YAML config like `constant_alpha: false` (yaml's false → Python False) intending "don't use a constant, use learnable α".

## Reproduction (nmn 0.2.30)

```python
import jax, jax.numpy as jnp
from flax import nnx
from nmn.nnx.layers import YatNMN

# Intent: learnable α (per docstring: "If None (default), use learnable alpha when use_alpha=True.")
layer = YatNMN(
    in_features=8, out_features=16,
    use_alpha=True,
    constant_alpha=False,        # ← user means "no constant"; nmn reads "constant 0.0"
    rngs=nnx.Rngs(0),
)
print("constant_alpha =", layer.constant_alpha)                  # False
print("_constant_alpha_value =", layer._constant_alpha_value)    # 0.0  ← bug

# 'alpha' is NOT in nnx.Param state — non-learnable
state = nnx.state(layer, nnx.Param)
print("param keys:", list(jax.tree_util.tree_leaves_with_path(state)))

# Forward: output is the constant-α path, multiplied by 0
x = jax.random.normal(jax.random.key(0), (1, 8))
y = layer(x)
print("output norm:", jnp.linalg.norm(y).item())                  # 0.0
```

## Source pointer

`src/nmn/nnx/layers/nmn.py` lines 226–233 (v0.2.30):

```python
if constant_alpha is not None:
    if constant_alpha is True:
        self._constant_alpha_value = self.DEFAULT_CONSTANT_ALPHA
    else:
        self._constant_alpha_value = float(constant_alpha)
    use_alpha = True
else:
    self._constant_alpha_value = None
    if use_alpha:
        # ... build learnable α
```

`False is not None` enters the outer branch, `False is True` is false so it falls to `float(False) == 0.0`. Same trap exists for `0`, `0.0`, and any falsy numeric.

## Suggested fix (smallest delta)

Treat `constant_alpha=False` as equivalent to `constant_alpha=None` — i.e., explicitly opt out of the constant path. This matches what the user almost certainly meant (and what the docstring's enumeration suggests):

```python
if constant_alpha is not None and constant_alpha is not False:
    if constant_alpha is True:
        self._constant_alpha_value = self.DEFAULT_CONSTANT_ALPHA
    else:
        self._constant_alpha_value = float(constant_alpha)
    use_alpha = True
else:
    self._constant_alpha_value = None
    ...
```

Or, more strictly: raise on `False` so the trap surfaces immediately:

```python
if constant_alpha is False:
    raise ValueError(
        "constant_alpha=False is ambiguous. Pass None for learnable α "
        "(when use_alpha=True), True for sqrt(2), or a float."
    )
```

Either way, the docstring should be updated to call out the `False` behavior explicitly.

## Real-world impact

This was caught downstream while porting flaxchat's GOAT_VVV to a new training harness. The model trained without errors but never actually used the YatNMN MLP — degenerate "embedding + zero-init attention only" model. Loss looked superficially finite, so it took a careful read of nmn's source to track down. A user who isn't reading nmn's `__init__` line-by-line is unlikely to spot it.

## Environment

- nmn 0.2.30 (PyPI)
- Python 3.13
- flax 0.12.x, jax 0.9.2
